web/src/reducers/dataReducer.js

```diff
@@ -53,6 +61,15 @@ const initialDataState = {
 
 module.exports = (state = initialDataState, action) => {
   switch (action.type) {
+    case 'APPLICATION_STATE_UPDATE': {
+      // Reset histories if timescale changes
+      const { key } = action;
+      if (key === 'timescale') {
+        return { ...state, histories: {} };
+      }
+      return state;
+    }
```

Comment:
This looks a little bit like an anti-pattern - I think it would make sense in this case to create a separate action `TOGGLE_TIMESCALE` that would both reset the histories and set the `timescale` key to prevent more logic piling up under the general `APPLICATION_STATE_UPDATE` action.

Reply:
Indeed I agree. However, I was hesitant to add a new action as this would mean there would be two ways to update the application state: 1/ APPLICATION_STATE_UPDATE 2/ TOGGLE_TIMESCALE.
However, only one of those two would have the right logic.

Any ideas?

Reply:
> Any ideas?

Yes, an alternative solution might be to add a new action `RESET_HISTORIES` here that would have to be called explicitly together with setting the `timescale` or alternatively create a `dispatchData` helper modelled by `dispatchApplication` for the same effect.

I know that these proposals make calling these actions a bit more complicated (two calls instead of one), but I think it pays better in a longer run to keep the reducers simple vs. branching into the cases in their logic.

Not a critical change though :stuck_out_tongue: